src/PatchOrchestrationApplication/CoordinatorService/src/RepairManagerHelper.cs

```diff
@@ -274,6 +279,156 @@ internal async Task PrintRepairTasks(CancellationToken cancellationToken)
             }
         }
 
+        public async Task PostRMTaskUpdates(CancellationToken cancellationToken)
+        {
+            try
+            {
+                NodeList nodeList = await this.fabricClient.QueryManager.GetNodeListAsync(null, null, this.DefaultTimeoutForOperation, cancellationToken);
+                IList<RepairTask> claimedTaskList = await this.GetClaimedRepairTasks(nodeList, cancellationToken);
+                RepairTaskList processingTaskList = await this.GetRepairTasksUnderProcessing(cancellationToken);
+                cancellationToken.ThrowIfCancellationRequested();
+                if (claimedTaskList.Any())
+                {
+                    if (!processingTaskList.Any())
+                    {
+                        // This means that repair tasks are not getting approved.
+                        ClusterHealth clusterHealth = await this.fabricClient.HealthManager.GetClusterHealthAsync();
+                        if (clusterHealth.AggregatedHealthState == HealthState.Error)
+                        { 
+                            // Reset Count
+                            postUpdateCount = 0;
+                            string warningDescription = "Cluster is unhealthy. POA Repair task created for OS update will not be approved. Please take cluster to healthy state for POA to start working.";
+                            HealthManagerHelper.PostNodeHealthReport(this.fabricClient, this.context.ServiceName, RMTaskUpdateProperty, warningDescription, HealthState.Ok, 5);
+                        }
+                        else
+                        {
+                            postUpdateCount++;
+                            if (postUpdateCount > 60)
+                            {
+                                // Reset Count and throw a warning on the service saying we dont know the reason. But POA not is not approving tasks.
+                                postUpdateCount = 0;
+                                string warningDescription = "POA repair tasks are not getting approved. So, update installation is halted. Please try to find out why is this blocked.";
```

Comment:
>POA repair tasks are not getting approved. So, update installation is halted. Please try to find out why is this blocked [](start = 61, length = 120)

Patch Orchestration Application is currently not patching nodes. This could be possible if there is some node which is stuck in disabling state for long time.